net/batching/conn_linux.go

```diff
@@ -328,10 +329,15 @@ func tryEnableUDPOffload(pconn nettype.PacketConn) (hasTX bool, hasRX bool) {
 			return
 		}
 		err = rc.Control(func(fd uintptr) {
-			_, errSyscall := syscall.GetsockoptInt(int(fd), unix.IPPROTO_UDP, unix.UDP_SEGMENT)
-			hasTX = errSyscall == nil
-			errSyscall = syscall.SetsockoptInt(int(fd), unix.IPPROTO_UDP, unix.UDP_GRO, 1)
-			hasRX = errSyscall == nil
+			var errSyscall error
+			if !envknob.Bool("TS_TUN_DISABLE_UDP_GSO") {
```

Comment:
`TS_TUN...` implies we're disabling it on the overlay, but this is an underlay UDP socket.

Also, this package is shared between magicsock and peer relay, so naming should be generic (TS_BATCHING_CONN...), or it should move up to magicsock.